src/core/util/options.js

```diff
@@ -172,7 +172,7 @@ strats.watch = function (parentVal: ?Object, childVal: ?Object): ?Object {
     }
     ret[key] = parent
       ? parent.concat(child)
-      : [child]
+      : Array.isArray(child) ? child : [child]
```

Comment:
@yyx990803 I've added a new test for merging multiple layers of extends (I wasn't sure if it was more appropriate in the watch tests or the extends tests) - this causes the issue where child is wrapped multiple times. I am happy to make your change however this second test case will fail with your recommendation but work with the initial fix.